crates/solvers/src/domain/dex/slippage.rs

```diff
@@ -95,72 +98,67 @@ impl Slippage {
         // internally casting its `BigInt` digits to a `i128` and unwrapping.
         // This means that there is a maximum of 38-digits of precision
         // (specifically, `i128::MAX.to_string().len() - 1`) allowed when
-        // rounding. So, in order to be pragmatic (and seing that 38 digits of
+        // rounding. So, in order to be pragmatic (and seeing that 38 digits of
         // precision is more than enough for slippage), first truncate the
-        // value to the maximum preicision and then round.
+        // value to the maximum precision and then round.
         Self(self.0.with_prec(38).round(arg as _))
     }
 }
 
-/// Token reference prices for a specified auction.
-pub struct Prices(pub HashMap<eth::TokenAddress, BigDecimal>);
-
-impl Prices {
-    pub fn new(prices: impl IntoIterator<Item = (eth::TokenAddress, auction::Price)>) -> Self {
-        Self(
-            prices
-                .into_iter()
-                .map(|(token, price)| (token, conv::ether_to_decimal(&price.0)))
-                .collect(),
-        )
-    }
-
-    /// Computes the set of reference prices for the specified auction.
-    pub fn for_auction(auction: &auction::Auction) -> Self {
-        Self::new(
-            auction
-                .tokens
-                .iter()
-                .filter_map(|(address, token)| Some((*address, token.reference_price?))),
-        )
-    }
-
-    /// Returns the reference price, in Ether for the specified token.
-    pub fn reference_price(&self, token: eth::TokenAddress) -> Option<auction::Price> {
-        self.0
-            .get(&token)
-            .and_then(conv::decimal_to_ether)
-            .map(auction::Price)
-    }
-}
-
 #[cfg(test)]
 mod tests {
-    use super::*;
+    use {
+        super::*,
+        crate::domain::auction::{self, Auction},
+    };
 
+    // TODO Notice that I'm forced to change the tests only because I refactored
+    // some code. None of the behavior changed, yet the tests are now useless
+    // because they've been changed.
+    // TODO Furthermore, notice how the domain changed to work with Auction as a
+    // consequence of gaining a deeper understanding of the domain (dex solving
+    // happens in the context of an auction, this is also clear since the dex
+    // solvers need multiple fields from the auction), and now the test breaks
+    // down. This is the opposite of what should happen. Nothing should ever
+    // hinder domain distillation.
```

Comment:
> We pass it in like that because it's easier at the place the function is used.

No, we pass it in like that because we're building coherent abstractions. `slippage::Prices` is not a coherent abstraction.

Reply:
> ...but now that we see that decimals are needed as well as the reference prices and the gas price from the auction, we have a pretty strong indicator that we should just use `Auction`.

Where in `Limits` do you need the decimals and gas price? IMO computing slippage is very far detached from the concept of an `Auction`. Only the reference prices happen to be in an auction but from the perspective of the `Limits` code it doesn't matter where the reference price comes from.

Reply:
My rationale for creating a `dex::slippage::Prices` utility type was specifically because slippage calculation does not *want* auction prices, but token exchange rates to Ether, so it is conceptually different from auction reference prices.

Furthermore, it felt icky from an abstraction point of view to pass in a complete auction for slippage computation when all you need from an auction are its reference prices for computing Ether exchange rates. Auctions include orders, liquidity, deadline, token metadata, etc. which aren't really needed for slippage computation (and in the function signature is unclear . This manifests itself in the various `Default::default()` calls when setting up the unit test 👇. What would be slightly cleaner here IMO would be to pass in `Auction::prices` and not require the utility type anymore.

The unit test was included here as it testing this from the `Solver` boundary is quite difficult. I agree that having refactors require annoying changes to unit tests is sub-optimal.